server/item/stack.go

```diff
@@ -264,6 +266,18 @@ func (s Stack) Enchantments() []Enchantment {
 	return e
 }
 
+// RepairCost returns the number of experience levels to add to the base level cost when repairing, combining, or
+// renaming this item with an anvil.
+func (s Stack) RepairCost() int {
+	return s.repairCost
+}
+
+// WithRepairCost returns the current Stack with the repair cost set to the passed value.
+func (s Stack) WithRepairCost(repairCost int) Stack {
+	s.repairCost = repairCost
+	return s
+}
```

Comment:
I would check if the underlying item is `Repairable`, similar to what `Stack.Damage()` does, to avoid adding repair costs to things that can't have one.